src/index.js

```diff
@@ -95,27 +95,28 @@ class ESLintWebpackPlugin {
         return;
       }
 
-      // Gather Files to lint
-      compilation.hooks.finishModules.tap(ESLINT_PLUGIN, (modules) => {
-        /** @type {string[]} */
-        const files = [];
+      /** @type {string[]} */
+      const files = [];
 
-        // @ts-ignore
-        for (const { resource } of modules) {
-          if (resource) {
-            const [file] = resource.split('?');
-
-            if (
-              file &&
-              !files.includes(file) &&
-              isMatch(file, wanted) &&
-              !isMatch(file, exclude)
-            ) {
-              files.push(file);
-            }
+      // @ts-ignore
+      // Add the file to be linted
+      compilation.hooks.succeedModule.tap(ESLINT_PLUGIN, ({ resource }) => {
+        if (resource) {
+          const [file] = resource.split('?');
+
+          if (
+            file &&
+            !files.includes(file) &&
+            isMatch(file, wanted) &&
+            !isMatch(file, exclude)
+          ) {
+            files.push(file);
           }
         }
+      });
```

Comment:
sorry
I can't understand why this fix can resolve current issue.  
In my opinion,this just collect the files in another time.

Reply:
agreed

Reply:
The goal is to group the changed files and execute lint only at the end

`succeedModule` its called for every file and watch mode just file changed
`finishModules` is called with all modules and is responsible for linting added files

Reply:
When hot-update happens,i works,But when the project is started ,all files will be linted
https://github.com/webpack-contrib/eslint-webpack-plugin/issues/75#issuecomment-774022336

Reply:
For this you can use `lintDirtyModulesOnly`